app/models/topic.rb

```diff
@@ -2,4 +2,7 @@ class Topic < ApplicationRecord
   validates :name, presence: true
 
   belongs_to :user
+  # has_and_belongs_to_many :collections
+  has_many :collection_topic_memberships
+  has_many :topics, through: :collection_topic_memberships
```

Comment:
Think this is left over from our chat.
You shoulnd't (can't?) mix _:has_and_belongs_to_many_ relationships with _:has_many_ and _:has_many through_ relationships. Use one of the other.